nlp_primitives/number_of_common_words.py

```diff
@@ -0,0 +1,53 @@
+from typing import Iterable
+
+import pandas as pd
+from featuretools.primitives.base import TransformPrimitive
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import IntegerNullable, NaturalLanguage
+
+from .constants import common_words_1000
+
+
+class NumberOfCommonWords(TransformPrimitive):
+    """Determines the number of common words in a string.
+    Description:
+        Given list of strings, determine the number of words that appear in a supplied wordbank.
+        The list of strings is case insensitive. The word banks should be all lower case strings.
+        If a string is missing, return `NaN`.
+
+    Examples:
+        >>> x = ['Hey! This is some natural language', 'bacon, cheesburger, AND, fries', 'I! Am. A; duck?']
+        >>> number_of_common_words = NumberOfCommonWords(word_set={'and', 'some', 'am', 'a', 'the', 'is', 'i'})
+        >>> number_of_common_words(x).tolist()
+        [2, 1, 3]
+    """
+    name = "number_of_common_words"
+    input_types = [ColumnSchema(logical_type=NaturalLanguage)]
+    return_type = ColumnSchema(logical_type=IntegerNullable, semantic_tags={'numeric'})
+
+    default_value = 0
+
+    def __init__(self, word_set=set(common_words_1000), delimiters_regex=r"[- \[\].,!\?;\n]"):
```

Comment:
I see we have the option to specify `delimiters_regex` here. We should mention it in the docstring as well, like we do the word bank, explain how it is used and add a doctest example that shows how to use it.

Reply:
Could add this and `word_set` to an `Args` section  of the docstring